src/main/java/uk/ac/ebi/eva/pipeline/io/readers/NonAnnotatedVariantsMongoReader.java

```diff
@@ -31,13 +34,29 @@
  */
 public class NonAnnotatedVariantsMongoReader extends MongoDbCursorItemReader {
 
-    public NonAnnotatedVariantsMongoReader(MongoOperations template, String collectionsVariantsName) {
-        super();
+    private static final String STUDY_KEY = VariantToDBObjectConverter.FILES_FIELD + "."
+            + VariantSourceEntryToDBObjectConverter.STUDYID_FIELD;
+
+    /**
+     * @param studyId Can be the empty string, meaning to bring all non-annotated variants in the collection.
+     *  If the studyId string is not empty, bring only non-annotated variants from that study. This parameter should
+     *  not be null in any case.
+     */
+    public NonAnnotatedVariantsMongoReader(MongoOperations template, String collectionsVariantsName, String studyId)
+            throws JobParametersInvalidException {
+        if (studyId == null) {
```

Comment:
this reader may be used without injection into a step, the usage with a validator is not required. I don't see much difference between an assertionError and a RuntimeException.